### PR TITLE
[runtime] Clean up symbol names to not pollute the global namespace.

### LINKED ISSUE #7565: Generic symbol names exported from libmono
https://bugzilla.xamarin.com/show_bug.cgi?id=25821

---

Work started in #1742 a while back.

### PR DESCRIPTION
**Pull request by alexrp**
Offending symbols found with:

    nm -D --defined-only libmonosgen-2.0.so | cut -d ' ' -f 3 | grep -v '^eg_\|mono_\|monoeg_\|wapi_\|ves_icall_\|sgen_\|mini_\|__jit_debug.*$' | sort

Fixes #7565.

---

Continuation of #1742.

### CODE REVIEW

File: mono/eglib/gunicode-win32-uwp.c [hunk 1]
```diff
@@ -11,24 +11,24 @@
 #define CODESET 1
 #include <windows.h>
 
-extern const char *my_charset;
+extern const char *eg_my_charset;
 static gboolean is_utf8;
```
Comment (vargaz) [inline mono/eglib/gunicode-win32-uwp.c]: I think a lot of these symbols should be made static instead of renaming them.

Comment (alexrp) [reply mono/eglib/gunicode-win32-uwp.c]: I made symbols `static` where doing so was possible without further code change (such as exposing a getter function). This one in particular is, surprisingly, used across multiple files in eglib.

**Comment by lambdageek** [comment]
Why is this necessary? we build with `-fvisibility=hidden` by default.

**Comment by vargaz** [comment]
This could be a problem for static libs which have no visibility.

**Comment by alexrp** [comment]
@monojenkins build failed

File: mono/eglib/gunicode-win32.c [hunk 2]
```diff
@@ -11,21 +11,21 @@
 #define CODESET 1
 #include <windows.h>
 
-extern const char *my_charset;
+extern const char *eg_my_charset;
```
[the remark this answers is not in the document]
Comment (alexrp) [reply mono/eglib/gunicode-win32.c]: Arguably a lot of these `extern` declarations in `.c` files do, but I didn't want to do that kind of refactoring in this PR. All I'm doing here is either renaming or slapping on a `static` where possible.

**Comment by alexrp** [comment]
Ping? Would be good to get this merged before too many conflicts with other PRs start popping up.

**Pull request merged**